apps/admin/src/hooks/useEditTable.ts

```diff
@@ -0,0 +1,28 @@
+import type { FormInstance } from 'antd';
+import { useState } from 'react';
+
+const useEditTable = <T extends { key: string }>(form: FormInstance) => {
+  const [editingKey, setEditingKey] = useState('');
+
+  const isEditing = (record: T) => record.key === editingKey;
+
+  const handleEdit = (record: Partial<T> & { key: string }) => {
+    form.setFieldsValue({
+      ...record,
+    });
+    setEditingKey(record.key);
+  };
+
+  const cancel = () => {
+    setEditingKey('');
+  };
+
+  return {
+    editingKey,
+    isEditing,
+    handleEdit,
+    cancel,
+  };
+};
```

Comment:
```suggestion
  const cancel = () => {
    setEditingKey('');
  };
  
  const register = () => {
    isEditing,
    handleEdit,
    cancel,
  };

  return {
    editingKey,
    isEditing,
    handleEdit,
    cancel,
    register
  };
};
```

`useEditTable` 훅이 단순히 TableView의 `Props`를 내려준다면 `register` 인터페이스를 만들어서 쉽게 주입할 수 있도록 해도 좋아요. (DX면에서) 해당 인터페이스가 발전된 형태는 react-hook-form의 register인데요. 해당 인터페이스에 대해 살펴보는 것도 추천드립니다

ref: https://react-hook-form.com/docs/useform/register

Reply:
좋은 피드백 감사합니다! register 인터페이스를 만들어서 주입하는 방식이 훨씬 더 깔끔하고 효율적일 것 같네요 😊👍